### DISCUSSION TITLE
How to use?

### DISCUSSION THREAD

**Discussion by drallcom3** [jelte/be.khepri.play.assetdelivery.addressables#2] [General]
How do I use this tool?

I have an addressables bundle from an external project that I would like to place in Play Asset Delivery.

I have three file: mybundle.bundle, catalog.json and catalog.hash.
The groups in the bundle have your new Play Asset Delivery schema.
I can load the catalog perfectly fine if I plopp it in streaming assets (as a test).

How do I go from there?
The Asset Delivery tool doesn't recognize the .bundle files.
Even then, what do I do with the catalog files?

Thank you in advance!

**Comment by jelte** [comment]
Hey drallcom3, 

it's been a busy few weeks for me (other things to work on).
I'm hoping to write the documentation this week(end) detailing how things work and release a first version.

for what you copy, I think you are missing a few files. During build Addressables will copy all files in `Library\com.unity.addressables\StreamingAssetsCopy` to `StreamingAssets`:
- settings.json `< addressables configurate`
- link.xml `< needed to avoid some code stripping`
- catalog.bundle `< this includes the catalog.json and catalog.hash, but is compressed to save space`
- mybundle.bundle `< your bundle(s)`
- buildlog* `< useless files, which you could probably ignore/remove`

how this package will work, is to add "Library\com.unity.addressables\StreamingAssetsCopy\...\mybundle.bundle" as a Android AssetPack to your build and remove it from `StreamingAssets` right before building the final aab file. The catalog, settings and link files should remain in the `StreamingAssets` these are generally small (less than 2MB)

As it stand now, addressables from an external project will work. 
The main reason for this is that the configuration for the asset packs is set on the addressables groups configuration, which I guess isn't in your base project. I simply haven't taking this scenario into account as this is not a requirement for the project I work on. 
But it shouldn't be to difficult to make that work. just running it in my head and the changes should be minor, simply make the source path (where are the addressables asset located) configurable (done while writing this) and have the asset pack config written to a file which will be exported with the addressables in the external project.

Created the task #3.

**Comment by drallcom3** [comment]
Thank you for your response. No hurry to do anything. I was just trying your asset out of curiosity.

A local addressables build has these files. It would probably work there, I didn't test it.
Usually you would use asset delivery if you have a few hundreds megabytes of assets in your project and at that size you're looking to have an extra project for those art assets.
